archive/zip.ksy

```diff
@@ -357,3 +356,4 @@ enums:
     0x5455: extended_timestamp
     0x7855: infozip_unix
     0x7875: infozip_unix_var_size
+doc-ref: https://github.com/LuaDist/zip/blob/b710806/proginfo/extrafld.txt
```

Comment:
Wrong indentation. This way (with no indent), it will be parsed as if it was at the top level, so it will actually overwrite this `doc-ref` at the beginning of the file, which is really meant to apply to the entire format:

https://github.com/kaitai-io/kaitai_struct_formats/blob/bc8098d6e43b574c98c2d83fc894f2ec0680a670/archive/zip.ksy#L33-L35

Check for yourself at [https://yaml-online-parser.appspot.com/](https://yaml-online-parser.appspot.com/?type=json&url=https%3A%2F%2Fraw.githubusercontent.com%2Fkaitai-io%2Fkaitai_struct_formats%2Fbc8098d6e43b574c98c2d83fc894f2ec0680a670%2Farchive%2Fzip.ksy). The _Output_ looks like this:

```js
{
  "meta": {
    // ...
  },
  "seq": [
    {
      "repeat": "eos", 
      "type": "pk_section", 
      "id": "sections"
    }
  ],
  "doc": "ZIP is a popular archive file format ...",
  "enums": {
    // ...
  },
  "doc-ref": "https://github.com/LuaDist/zip/blob/b710806/proginfo/extrafld.txt",
  // ...
}
```

Note that

```json
  "doc-ref": [
    "https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT", 
    "https://users.cs.jmu.edu/buchhofp/forensics/formats/pkzip.html"
  ],
```

is nowhere to be found - it has been silently overwritten by the later `doc-ref` key on the top level (i.e. by the `doc-ref: https://github.com/LuaDist/zip/blob/b710806/proginfo/extrafld.txt`). Most YAML parsers allow duplicate keys and they keep only one of them (either the last or first one). However, as of 0.9, the JVM version of the `kaitai-struct-compiler` (which uses SnakeYAML as the YAML parser under the hood) even refuses any KSY spec that contains duplicate keys, if I'm not mistaken. So this spec `zip.ksy` would no longer compile.

---

Anyway, `doc` and `doc-ref` are not supported currently for an enum as a whole (see https://github.com/kaitai-io/kaitai_struct/issues/358), so I suggest that you put it above the enum declaration (as it seems common in the programming world, see https://github.com/kaitai-io/kaitai_struct_formats/pull/479#discussion_r616522477). Like this:

```diff
+  # https://github.com/LuaDist/zip/blob/b710806/proginfo/extrafld.txt
   extra_codes:
-    # https://github.com/LuaDist/zip/blob/b710806/proginfo/extrafld.txt
     0x0001: zip64
     0x0007: av_info
```

IMHO it makes slightly more sense than the previous version.